core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -119,35 +131,52 @@ class CloudSqlInstance {
 
   /**
    * Initializes a new Cloud SQL instance based on the given connection name.
-   *
-   * @param connectionName instance connection name in the format "PROJECT_ID:REGION_ID:INSTANCE_ID"
-   * @param apiClient      Cloud SQL Admin API client for interacting with the Cloud SQL instance
+   *  @param key instance connection name in the format "PROJECT_ID:REGION_ID:INSTANCE_ID"
+   * @param apiClientOpt      Cloud SQL Admin API client for interacting with the Cloud SQL instance
+   * @param credentialFactoryOpt     Cloud SQL Admin API client credential factory
+   * @param serverProxyPort proxy port
    * @param executor       executor used to schedule asynchronous tasks
    * @param keyPair        public/private key pair used to authenticate connections
    */
   CloudSqlInstance(
-      String connectionName,
-      SQLAdmin apiClient,
-      boolean enableIamAuth,
-      CredentialFactory tokenSourceFactory,
+      CloudSqlInstanceKey key,
+      Optional<SQLAdmin> apiClientOpt,
+      Optional<CredentialFactory> credentialFactoryOpt,
```

Comment:
Are these optional for testing purposes?

Reply:
Yes. This keeps the mock tests working. Normally, the CloudSqlInstance is responsible for creating its own CredentialFactory and SqlAdmin